README.md

```diff
@@ -0,0 +1,182 @@
+B: That will make him so happy.
+A: Yeah, we’ve discussed it many times. I think he’s ready now.
+B: That’s good. Raising a dog is a tough issue. Like having a baby ;-)
+A: I'll get him one of those little dogs.
+```
+
+This is because nothing in the input indicated that the conversation was over, and that the model
+should begin producing the summary. If you were using a model such as Vicuna, the inherent 
+PromptFormat would solve this problem:
+
+```
+User: Please summarize the following conversation
+A: Hi Tom, are you busy tomorrow’s afternoon?
+B: I’m pretty sure I am. What’s up?
+A: Can you go with me to the animal shelter?.
+B: What do you want to do?
+A: I want to get a puppy for my son.
+
+Assistant: Tom and his friend are going to the animal shelter to get a puppy for Tom's son.
+```
+
+The presence of the "Assistant:" string indicates to the language model that it's time to produce 
+the summary. 
+
+A better approach would be to bake this information into the Prompt object, so that the format
+of the model is irrelevant.
+
+```python
+Prompt(
+  system_message="Please summarize the following conversation", 
+  input_template="Conversation: {text}",
+  response_template="Summary: {text}"
+)
+```
+
+This would result in the following piece of training data when used with the base llama-2 model:
+
+```
+Please summarize the following conversation
+Conversation: A: Hi Tom, are you busy tomorrow’s afternoon?
+B: I’m pretty sure I am. What’s up?
+A: Can you go with me to the animal shelter?.
+B: What do you want to do?
+A: I want to get a puppy for my son.
+
+Summary: Tom and his friend are going to the animal shelter to get a puppy for Tom's son.
+```
+
+Or, if used with Vicuna:
+
+```
+Please summarize the following conversation
+User: Conversation: A: Hi Tom, are you busy tomorrow’s afternoon?
+B: I’m pretty sure I am. What’s up?
+A: Can you go with me to the animal shelter?.
+B: What do you want to do?
+A: I want to get a puppy for my son.
+
+Assistant: Summary: Tom and his friend are going to the animal shelter to get a puppy for Tom's son.
+```
+
+### Configuration
+
+We've written scripts for tasks (such as fine tuning, batch inference, model serving)
+So that you can ideally run these tasks without having to write any code at all. To do so we
+rely on a lightweight yaml configuration to direct the specifics of each task. 
+
+> [!NOTE]
+> Sometimes the configuration delegates to other code/classes, for example
+> - load_datest, load_from_disk for referencing HuggingFace datasets
+> - UserAssistantFormat, VicunaFormat for PromptFormats
+> - ZeroShotQA and FewShotQA classes for Prompts
+>
+> These configurations are specified with a `method` and a `kwargs` value. `method` is a string that
+has been registered against an existing python function in code. You can also call methods that
+we haven't registered with the following syntax: `path.to.module::name`. The kwargs entry is a dictionary of
+> parameters that the method expects.
```

Comment:
Might be nice to also support `args` list. That way load_dataset and load_from_disk could both be used the same way.

Reply:
yea i thnk that's a good idea - will save for the future